plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/polaris/PolarisHiveMetastore.java

```diff
@@ -0,0 +1,575 @@
+
+    public Map<String, PartitionStatistics> getPartitionStatistics(String databaseName, String tableName, Set<String> partitionNames)
+    {
+        return ImmutableMap.of();
+    }
+
+    @Override
+    public void updateTableStatistics(String databaseName, String tableName, OptionalLong acidWriteId, StatisticsUpdateMode mode, PartitionStatistics statisticsUpdate)
+    {
+        // Statistics updates not supported yet
+    }
+
+    @Override
+    public void updatePartitionStatistics(Table table, StatisticsUpdateMode mode, Map<String, PartitionStatistics> partitionUpdates)
+    {
+        // Statistics updates not supported yet
+    }
+
+    public List<String> getAllTables(String databaseName)
+    {
+        try {
+            // Get both Iceberg and generic tables
+            List<String> icebergTables = polarisClient.listTables(databaseName)
+                    .stream()
+                    .map(PolarisTableIdentifier::getName)
+                    .collect(toImmutableList());
+
+            List<String> genericTables = polarisClient.listGenericTables(databaseName)
+                    .stream()
+                    .map(PolarisTableIdentifier::getName)
+                    .collect(toImmutableList());
+
+            return ImmutableList.<String>builder()
+                    .addAll(icebergTables)
+                    .addAll(genericTables)
+                    .build();
+        }
+        catch (Exception e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, "Failed to list tables in database: " + databaseName, e);
+        }
+    }
+
+    @Override
+    public List<TableInfo> getTables(String databaseName)
+    {
+        return getAllTables(databaseName).stream()
+                .map(tableName -> new TableInfo(new SchemaTableName(databaseName, tableName), TableInfo.ExtendedRelationType.TABLE))
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public List<String> getTableNamesWithParameters(String databaseName, String parameterKey, Set<String> parameterValues)
+    {
+        return ImmutableList.of();
+    }
+
+    public List<String> getAllViews(String databaseName)
+    {
+        try {
+            return polarisClient.listViews(databaseName)
+                    .stream()
+                    .map(PolarisTableIdentifier::getName)
+                    .collect(toImmutableList());
+        }
+        catch (Exception e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, "Failed to list views in database: " + databaseName, e);
+        }
+    }
+
+    @Override
+    public void createDatabase(Database database)
+    {
+        try {
+            PolarisNamespace namespace = new PolarisNamespace(database.getDatabaseName(), database.getParameters());
+            polarisClient.createNamespace(namespace);
+        }
+        catch (Exception e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, "Failed to create database: " + database.getDatabaseName(), e);
+        }
```

Comment:
For all of the cases that look like this, I think you can reduce `Exception` to `RuntimeException`